chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -206,44 +203,40 @@ type filteredBlock struct {
 func (n *NeutrinoNotifier) onFilteredBlockConnected(height int32,
 	header *wire.BlockHeader, txns []*btcutil.Tx) {
 
-	// Append this new chain update to the end of the queue of new chain
-	// updates.
-	n.newBlocksMtx.Lock()
-	n.newBlocks = append(n.newBlocks, &filteredBlock{
+	newBlock := filteredBlock{
 		hash:   header.BlockHash(),
 		height: uint32(height),
 		txns:   txns,
-	})
-	n.newBlocksMtx.Unlock()
-
-	// Launch a goroutine to signal the notification dispatcher that a new
-	// transaction update is available. We do this in a new goroutine in
-	// order to avoid blocking the main loop of the rescan.
-	go func() {
-		n.newBlocksUpdateSignal <- struct{}{}
-	}()
+	}
+
+	// Append this new chain update to the end of the queue of new chain
+	// updates.
+	select {
+	case n.newBlocks <- &newBlock:
+	default:
+		chainntnfs.Log.Errorf("New blocks buffered channel has reached "+
```

Comment:
This is potentially dangerous. Notifications should _never_ be dropped, and _must_ be delivered in order. Also the size of `1024` is insufficient as upon restart, the backing chain source may itself be syncing from before the chain tip possibly causing hundreds of blocks to be attached at once. 

As a result, this approach won't be accepted in it's current state. 

Instead, I've been meaning on refactoring this section to use [`sync.Cond`](https://golang.org/pkg/sync/#Cond), using it to implement a good ol' synchronized queue.

Reply:
Why is that preferable to using a buffered channel, which I'm pretty sure acts exactly like a synchronized queue? If dropping blocks is unacceptable, we could raise capacity and make this a blocking push into the channel (and it should never really block ideally).

Reply:
The synchronized queue can have an unbounded length, while the length of the buffered channel must be declared an initialization. 

The callback itself can't block as in the `btcd` case, it's executed synchronously from the main web-sockets processing loop. We share a websockets connection in a few places with other sub-systems, so that may impede their operation. Within`neutrino` it'll similarly halt the rescan loop.

Reply:
Replaced that commit with https://github.com/lightningnetwork/lnd/pull/360.